lib/client/pool_e2e_test.go

```diff
@@ -60,27 +60,27 @@ var _ = Describe("Pool tests", func() {
 				log.Println("Error creating client:", err)
 			}
 			By("Updating the pool before it is created")
-			_, outError := c.Pools().Update(&api.Pool{Metadata: meta1, Spec: spec1})
+			_, outError := c.IPPools().Update(&api.IPPool{Metadata: meta1, Spec: spec1})
 
 			// Should return an error.
 			Expect(outError.Error()).To(Equal(errors.New("resource does not exist: Pool(cidr=10.0.0.0/24)").Error()))
```

Comment:
one last thing, if you end up changing https://github.com/projectcalico/libcalico-go/pull/240#discussion-diff-86075055R60, this error assertion would also change along with Line 147 in this file